src/components/PurchaseBook.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {useState, useContext} from 'react';
+import {Col, Row, Typography, Select, Button, Divider, message} from 'antd';
+import {Book} from '../models/Book';
+import { BasketContext } from '../state/basket/BasketContext';
+
+const { Option } = Select;
+const {Title, Text} = Typography;
+
+interface PurchaseProps{
+    book: Book
+}
+
+const PurchaseBook = ({book}: PurchaseProps) => {
+    const {basket, addToBasket} = useContext(BasketContext);
+    const isInStock = book.stockAmount > 0;
+
+    const maxQuantity = book.stockAmount < 20 ? book.stockAmount : 20;
+    const numRange = new Array<number>(maxQuantity);
+    for(let i = 0; i < maxQuantity; i++){
+        numRange.push(i+1);
```

Comment:
You are already doing the work of looping here. I would suggest adding the jsx elements to the array.

```
numRange.push(<Option key={num} value={num}>{i+1}</Option>);
```

And the below:
```
Select quantity:         
<Select defaultValue={1} onChange={handleQuantitySelect}>
   {numRange}
</Select>
```

But obviously rename numRange to something that makes more sense like `dropdownQuantityItems`